### PR TITLE
add limit data in interviewStats page

### PR DESCRIPTION
**Pull request by ada-only-one-ada**

### CODE REVIEW

File: src/api/routes/interviews.ts [hunk 1]
```diff
@@ -186,11 +186,12 @@ const listInterviewerStats = procedure
 .output(z.array(z.object({
   user: zUser,
   interviews: z.number(),
+  limit: zInterviewerPreference.shape.limit,
```
Comment (weihanwang) [inline src/api/routes/interviews.ts:189]: why not just return the whole preference object?

File: src/pages/interviewers.tsx [hunk 2]
```diff
@@ -53,14 +56,19 @@ export default function Page() {
...
 }) {
   const role = isPermitted(user.roles, 'MentorCoach')
     ? 'MentorCoach' : isPermitted(user.roles, 'Mentor')
-    ? 'Mentor' : null;
-  const roleColorScheme = role == 'MentorCoach' ? "yellow" : "teal";
+    ? 'Mentor' : isPermitted(user.roles, 'Interviewer')
+    ? 'Interviewer' : null;
+  const roleColorScheme = role == 'MentorCoach' 
+    ? "yellow" : role == 'Mentor'
+    ? "teal" : "brand";
```
Comment (weihanwang) [inline src/pages/interviewers.tsx:71]: undo this block of change. we don't need to identify interviewer role.

File: src/pages/interviewers.tsx [hunk 3]
```diff
@@ -70,6 +78,10 @@ function Row({ user, interviews }: {
       </Tag>}
     </Td>
     <Td>{interviews}</Td>
+    <Td>
+      {noMoreThan !== undefined && until !== undefined 
+        ? `${noMoreThan} 直到 ${until.slice(0, 10)}` : ''}
```
Comment (weihanwang) [inline src/pages/interviewers.tsx:83]: why not simply `&& ...}` and remove `: ''`

Comment (ada-only-one-ada) [reply src/pages/interviewers.tsx:83]: that would remove entire column if no one sets limit.

File: src/pages/interviewers.tsx [hunk 4]
```diff
@@ -53,14 +55,17 @@ export default function Page() {
...
   user: User,
   interviews: number,
+  preference: UserPreference | null,
 }) {
   const role = isPermitted(user.roles, 'MentorCoach')
     ? 'MentorCoach' : isPermitted(user.roles, 'Mentor')
     ? 'Mentor' : null;
   const roleColorScheme = role == 'MentorCoach' ? "yellow" : "teal";
+  const until = preference?.interviewer?.limit?.until;
+  const noMoreThan = preference?.interviewer?.limit?.noMoreThan;
```
Comment (weihanwang) [inline src/pages/interviewers.tsx:68]: change to `const limit = preference?.interviewer?.limit`

File: src/pages/interviewers.tsx [hunk 5]
```diff
@@ -70,6 +75,10 @@ function Row({ user, interviews }: {
       </Tag>}
     </Td>
     <Td>{interviews}</Td>
+    <Td>
+      {noMoreThan !== undefined && until !== undefined
```
Comment (weihanwang) [inline src/pages/interviewers.tsx:79]: change to {limit && `${limit.noMoreThan} 直到 ${limit.until.slice(0, 10)}`. also comment on why `slice`

**Review by weihanwang — APPROVED** [review]

**Pull request merged**